pkg/imgpkg/api/describe_bundle.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package api
```

Comment:
should we move this into bundle package?

Reply:
I thought about that but my objective was to create a clear interface that we could consider stable and tell people that they can use when interacting with imgpkg as a library.

Reply:
> tell people that they can use when interacting with imgpkg as a library.

i don't doubt that some users only need this feature (we know of one consumer), but entirety of pkg/imgpkg/bundle and pkg/imgpkg/registry is the client library i would expect consumers to use.